File: r/scripts/read/read_tradingview_excel.R
#!/usr/bin/env Rscript
# Read TradingView Excel export and analyze the data

library(readxl)
library(dplyr)
library(lubridate)

# Read Excel file
excel_file <- "C:/Users/ROG/Downloads/三日暴跌接针策略_BINANCE_PEPEUSDT_2025-10-27_df79a.xlsx"

cat("Reading Excel file:", excel_file, "\n")

# Get sheet names
sheet_names <- excel_sheets(excel_file)
cat("Available sheets:", paste(sheet_names, collapse=", "), "\n\n")

# Read first sheet
df <- read_excel(excel_file, sheet = sheet_names[1])

cat("Data loaded successfully!\n")
cat("Dimensions:", nrow(df), "rows x", ncol(df), "columns\n\n")

cat("Column names:\n")
for(i in 1:ncol(df)) {
  cat(sprintf("  %d: %s\n", i, colnames(df)[i]))
}

cat("\nFirst 10 rows:\n")
print(head(df, 10))

cat("\nLast 5 rows:\n")
print(tail(df, 5))

cat("\nData structure:\n")
str(df)

cat("\nSummary statistics:\n")
print(summary(df))

# Save to CSV
output_csv <- "data/tradingview_results.csv"
write.csv(df, output_csv, row.names = FALSE, fileEncoding = "UTF-8")
cat("\nSaved to CSV:", output_csv, "\n")

# Save detailed info
output_info <- "docs/reports/tradingview_info.txt"
sink(output_info)
cat("TradingView Results Analysis\n")
cat(paste(rep("=", 80), collapse=""), "\n\n")
cat("Shape:", nrow(df), "rows x", ncol(df), "columns\n\n")
cat("Column names:\n")
for(i in 1:ncol(df)) {
  cat(sprintf("  %d: %s\n", i, colnames(df)[i]))
}
cat("\nData structure:\n")
str(df)
cat("\nFirst 20 rows:\n")
print(head(df, 20))
cat("\nSummary statistics:\n")
print(summary(df))
sink()

cat("Saved detailed info to:", output_info, "\n")

# Try to extract key metrics if available
cat("\n" , paste(rep("=", 80), collapse=""), "\n")
cat("EXTRACTING KEY METRICS\n")
cat(paste(rep("=", 80), collapse=""), "\n\n")

# Look for performance summary columns
possible_metric_cols <- c("Net Profit", "Total Trades", "Percent Profitable",
                          "Max Drawdown", "Return", "Win Rate", "Total Closed Trades",
                          "Profit Factor", "Sharpe Ratio")

for(col in possible_metric_cols) {
  if(col %in% colnames(df)) {
    cat(sprintf("%s: %s\n", col, df[[col]][1]))
  }
}

# Check if this is a trade list
if("Entry Time" %in% colnames(df) || "Exit Time" %in% colnames(df)) {
  cat("\nThis appears to be a TRADE LIST\n")
  cat("Number of trades:", nrow(df), "\n")

  # Show first 20 trades
  cat("\nFirst 20 trades:\n")
  print(head(df, 20))
}

cat("\nDone!\n")

File: python/scripts/read_tradingview_excel.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""
Read TradingView Excel export and analyze the data
"""

import sys

try:
    import pandas as pd
    print("pandas imported successfully")
except ImportError:
    print("ERROR: pandas not available")
    sys.exit(1)

# Try to read Excel file
excel_file = r'C:\Users\ROG\Downloads\三日暴跌接针策略_BINANCE_PEPEUSDT_2025-10-27_df79a.xlsx'

print(f"Attempting to read: {excel_file}")

try:
    # Try with openpyxl engine first
    df = pd.read_excel(excel_file, engine='openpyxl')
    print("Successfully read with openpyxl engine")
except Exception as e1:
    print(f"openpyxl failed: {e1}")
    try:
        # Try with xlrd engine
        df = pd.read_excel(excel_file, engine='xlrd')
        print("Successfully read with xlrd engine")
    except Exception as e2:
        print(f"xlrd failed: {e2}")
        try:
            # Try without specifying engine
            df = pd.read_excel(excel_file)
            print("Successfully read without specifying engine")
        except Exception as e3:
            print(f"Default engine failed: {e3}")
            print("\nTrying to read all available sheets...")
            try:
                xl_file = pd.ExcelFile(excel_file)
                print(f"Available sheets: {xl_file.sheet_names}")
                df = pd.read_excel(xl_file, sheet_name=xl_file.sheet_names[0])
                print(f"Successfully read first sheet: {xl_file.sheet_names[0]}")
            except Exception as e4:
                print(f"All methods failed: {e4}")
                sys.exit(1)

print(f"\nDataFrame loaded successfully!")
print(f"Shape: {df.shape}")
print(f"\nColumn names:")
for i, col in enumerate(df.columns):
    print(f"  {i}: {col}")

print(f"\nFirst 10 rows:")
print(df.head(10))

print(f"\nLast 5 rows:")
print(df.tail(5))

print(f"\nData types:")
print(df.dtypes)

print(f"\nBasic statistics:")
print(df.describe())

# Save to CSV for easier inspection
output_csv = r'data/tradingview_results.csv'
df.to_csv(output_csv, index=False, encoding='utf-8-sig')
print(f"\nSaved to: {output_csv}")

# Also save detailed info
output_info = r'docs/reports/tradingview_info.txt'
with open(output_info, 'w', encoding='utf-8') as f:
    f.write(f"TradingView Results Analysis\n")
    f.write(f"="*80 + "\n\n")
    f.write(f"Shape: {df.shape}\n")
    f.write(f"Rows: {len(df)}\n")
    f.write(f"Columns: {len(df.columns)}\n\n")
    f.write(f"Column names:\n")
    for i, col in enumerate(df.columns):
        f.write(f"  {i}: {col}\n")
    f.write(f"\n")
    f.write(f"Data types:\n{df.dtypes}\n\n")
    f.write(f"First 20 rows:\n{df.head(20)}\n\n")
    f.write(f"Basic statistics:\n{df.describe()}\n")

print(f"Saved detailed info to: {output_info}")
